github/util.py

```diff
@@ -417,22 +423,62 @@ def create_tag(
             tagger=author
         )
 
+    def _replacement_release_notes(
+        self,
+        asset_url: str,
+        component_name: str,
+        component_version: str,
+    ):
+        return (
+            f'The release-notes for component **{component_name}** in version '
+            f'**{component_version}** exceeded the maximum length allowed by GitHub for '
+            'release-bodies.\n'
+            f'They have been uploaded as release-asset and can be found at {asset_url}.'
+        )
+
+    RELEASE_NOTES_ASSET_NAME = 'release_notes.md'
+
     def create_release(
         self,
         tag_name: str,
         body: str,
         draft: bool=False,
         prerelease: bool=False,
-        name: str=None
+        name: str=None,
+        component_name: str=None,
+        component_version: str=None,
     ):
-        release = self.repository.create_release(
-            tag_name=tag_name,
-            body=body,
-            draft=draft,
-            prerelease=prerelease,
-            name=name
-        )
-        return release
+        if len(body) < MAXIMUM_GITHUB_RELEASE_BODY_LENGTH:
+            return self.repository.create_release(
+                tag_name=tag_name,
+                body=body,
+                draft=draft,
+                prerelease=prerelease,
+                name=name
+            )
+        else:
+            release = self.repository.create_release(
```

Comment:
add a brief comment as to why we create the release w/ empty body, to edit it afterwards (reason being we need to know the asset-url, which we cannot know (can we not actually?) w/o previously uploading the asset (which we cannot do w/o an existing release).

to be a bit more on the safe side, we might consider to firstly create the release as a draft, and then promote it to final after asset-upload (or purge it upon errs) - wdyt?